core/store/src/main/java/mil/nga/giat/geowave/core/store/query/QueryOptions.java

```diff
@@ -379,7 +407,27 @@ public static long getSerialversionuid() {
 			}
 		}
 
-		final ByteBuffer buf = ByteBuffer.allocate(16 + authBytes.length + aSize + iSize);
+		int fieldIdsBytesNeeded = 0;
```

Comment:
You used org.apache.commons.lang3.StringUtils.join else where.  Could we use that here?